src/ui/public/courier/fetch/call_client.js

```diff
@@ -76,17 +82,30 @@ export function CallClientProvider(Private, Promise, esAdmin, es) {
       });
     });
 
-
     // Now that all of THAT^^^ is out of the way, lets actually
     // call out to elasticsearch
-    Promise.map(executable, function (req) {
-      return Promise.try(req.getFetchParams, void 0, req)
-      .then(function (fetchParams) {
-        return (req.fetchParams = fetchParams);
+    promiseMapSettled(
+      executable,
+      function (request) {
+        return Promise.try(request.getFetchParams, void 0, request)
+        .then(function (fetchParams) {
+          return (request.fetchParams = fetchParams);
+        });
+      },
+      Promise)
+    .then(function (results) {
+      const requestsWithFetchParams = results.map((result, index) => {
+        if (result.resolved) {
+          return result.resolved;
+        } else {
+          const request = executable[index];
+          request.handleFailure(result.rejected);
+          executable[index] = undefined;
+        }
       });
-    })
-    .then(function (reqsFetchParams) {
-      return strategy.reqsFetchParamsToBody(reqsFetchParams);
+      _.remove(executable, request => request === undefined);
```

Comment:
Can we use `x = x.filter(...)` so that both `executable` and `requestsWithFetchParams` have to be defined with `let` and signify that they are modified below.